### DISCUSSION TITLE
chirpy adding blog name in the <title> tag. Making it more than 70 characters

### DISCUSSION THREAD

**Discussion by qureshiaquib** [cotes2020/jekyll-theme-chirpy#1833] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

I've found in bing indexer that most of my post is 70 characters and long hence indexing would be an issue. this is of critical high priority item. I dig down on my post title and i found in the html tag of `<title>` there is an addition of my blog name in every blogpost title tag.
for example my blog title is `"Access Azure File Share from Remote sites via Entra ID joined devices"`
but when i see title tag then i see 
`<title>Access Azure File Share from Remote sites via Entra ID joined devices | Azure Doctor</title>`

in this Azure Doctor is my blog name.

is there is way to disable this behaviour. Or i have to account the blog name and make my post smaller to reduce it to 70 characters?

### Operations you have already tried

checked the html code and found the title tag.

**Comment by kungfux** [comment]
> is there is way to disable this behaviour. Or i have to account the blog name and make my post smaller to reduce it to 70 characters?

This is a typical pattern for site titles, I would say, to append site name to the title. So, yeah, just account for that, or override the behavior in [_includes/head.html](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/_includes/head.html#L53) if you prefer to do so.

**Reply by qureshiaquib** [reply]
hey thanks Alex, I'm not a developer but just tried the html
Can i replace this

```
 <title>
    {%- unless page.layout == 'home' -%}
      {{ page.title | append: ' | ' }}
    {%- endunless -%}
    {{ site.title }}
  </title>

```

with this 

```
<title>
  {%- unless page.layout == 'home' -%}
    {{ page.title }}
  {%- endunless -%}
</title>
```

**Reply by kungfux** [reply, marked as answer]
Yeah, it looks good 👍

Please be aware that you will have to maintain the overridden version of the html once you upgrade to the newer version of the Chirpy if there will be some changes made to that file.

**Reply by qureshiaquib** [reply]
yes. I have done little customization here and there. and I'm planning to document this.

**Reply by qureshiaquib** [reply]
thanks for all the help Alex

**Reply by qureshiaquib** [reply]
Hi Alex, for some reason i don't find this file in the chirpy starter pack.
_includes/head.html

Also if I goto _Site/post/actualpost/index.html
and remove "| Azure Doctor" from each post, when i build exec it and see, it gets added automatically it seems.

**Reply by kungfux** [reply]
> Hi Alex, for some reason i don't find this file in the chirpy starter pack.
> _includes/head.html

That's expected. This file can be found in this repository only. Just follow the link I've shared previously.

> Also if I goto _Site/post/actualpost/index.html
> and remove "| Azure Doctor" from each post, when i build exec it and see, it gets added automatically it seems.

That's expected. You build the site from markdown files into HTML as output to `_site` directory, so every time you build a site, the content of that folder will overwrite.

**Reply by qureshiaquib** [reply]
thanks for prompt response Alex, I'm little lost. 
here is my setup: 
I use chirpy starter pack with github pages. I've forked it in my personal repo and have clone it on my desktop. and then i push this through github actions so that build happens on github.

when i open my forked starter pack i don't see this file there. hence i was asking what is the process to include the "_includes/head.html" in my starter repo? and what all changes i would have to do apart from adding this file?

**Reply by kungfux** [reply]
Chirpy solution is being packed into Gem file and reused in Chirpy starter pack, that's why you can't see these files there. However, if you copy the file and put it into your repository to the same location, it will override the file from Gem. Like in this example, you don't have `_includes/head.html` meaning you use the default one from Gem but if you copy it and add to your repo, you will be able to modify it according to your needs.

**Reply by qureshiaquib** [reply]
hey thanks Alex, i tried it yesterday and it worked. thanks for all the help.
